### PR TITLE
Update samples and tests to target netcoreapp2.1

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
🆙 📅

File: test/ServerComparison.FunctionalTests/Helpers.cs [hunk 1]
```diff
@@ -45,5 +45,36 @@ public static string GetConfigContent(ServerType serverType, string iisConfig, s
...
+#else
+#error Tests targeting CLR must be compiled only on desktop.
+#endif
+            }
+            else if (runtimeFlavor == RuntimeFlavor.CoreClr)
+            {
+                targetFramework =
+#if NETCOREAPP2_0
+                "netcoreapp2.0";
+#elif NETCOREAPP2_1 || NET461
```
Comment (jkotalik) [inline test/ServerComparison.FunctionalTests/Helpers.cs]: Why do you check for net461 here?

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs [hunk 2]
```diff
@@ -227,8 +242,7 @@ private static Task CheckAppCompressionAsync(HttpClient client, ILogger logger)
         private static string GetContentLength(HttpResponseMessage response)
         {
             // Don't use response.Content.Headers.ContentLength, it will dynamically calculate the value if it can.
-            IEnumerable<string> values;
-            return response.Content.Headers.TryGetValues(HeaderNames.ContentLength, out values) ? values.FirstOrDefault() : null;
+            return response.Content.Headers.TryGetValues(HeaderNames.ContentLength, out var values) ? values.FirstOrDefault() : null;
```
Comment (jkotalik) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: If we are using c#7 for this repository, we can ignore values like
``` out _```

File: test/ServerComparison.FunctionalTests/Helpers.cs [hunk 3]
```diff
@@ -45,5 +45,36 @@ public static string GetConfigContent(ServerType serverType, string iisConfig, s
...
             return content;
         }
+
+        public static string GetTargetFramework(RuntimeFlavor runtimeFlavor)
+        {
+            string targetFramework = null;
+            if (runtimeFlavor == RuntimeFlavor.Clr)
+            {
+#if NET461
+                targetFramework = "net461";
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/Helpers.cs]: `return "net461"`

File: test/ServerComparison.FunctionalTests/Helpers.cs (hunk 1 above)
Comment (pranavkm) [reply test/ServerComparison.FunctionalTests/Helpers.cs]: Desktop will continue to run tests for `RuntimeFlavor.CoreCLR` (I haven't opted out of those). We need to pick some TFM to run the server in this case.

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs (hunk 2 above)
Comment (pranavkm) [reply test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: We're using the variable.

**Pull request closed without merging**